File: web-front-reactive/src/main/java/template/modules/admin/command/users/AddUserCommand.java
package template.modules.admin.command.users;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;
import template.modules.admin.domain.users.User;
import template.modules.admin.domain.users.UserRepository;

@Service
@AllArgsConstructor
public class AddUserCommand {
    private final UserRepository repo;

    public Mono<User> execute(User user) {
        return repo.save(user);

    }

}

File: web-front-reactive2/src/main/kotlin/template/modules/admin/command/users/AddUserCommand.kt
package template.modules.admin.command.users

import kotlinx.coroutines.reactor.awaitSingleOrNull
import org.springframework.stereotype.Service
import template.modules.admin.domain.users.User
import template.modules.admin.domain.users.UserRepository

@Service
class AddUserCommand(private val repo: UserRepository) {
    suspend fun execute(user: User) = repo.save(user).awaitSingleOrNull();

}